src/highdicom/spatial.py

```diff
@@ -35,6 +35,395 @@ def create_rotation_matrix(
+        origin of the reference coordinate system along the X, Y, and Z
+        axis
+    image_orientation: Sequence[float]
+        Cosines of the row direction (first triplet: horizontal, left to
+        right, increasing column index) and the column direction (second
+        triplet: vertical, top to bottom, increasing row index) direction
+        expressed in the three-dimensional patient or slide coordinate
+        system defined by the Frame of Reference
+    pixel_spacing: Sequence[float]
+        Spacing between pixels in millimeter unit along the column
+        direction (first value: spacing between rows, vertical, top to
+        bottom, increasing row index) and the rows direction (second value:
+        spacing between columns: horizontal, left to right, increasing
+        column index)
+    spacing_between_slices: float, optional
+        Distance (in the coordinate defined by the Frame of Reference)
+        between neighboring slices. Default: 1
+
+    Raises
+    ------
+    TypeError
+        When `image_position`, `image_orientation`, or `pixel_spacing` is
+        not a sequence.
+    ValueError
+        When `image_position`, `image_orientation`, or `pixel_spacing` has
+        an incorrect length.
+
+    """
+    if not isinstance(image_position, Sequence):
+        raise TypeError('Argument "image_position" must be a sequence.')
+    if len(image_position) != 3:
+        raise ValueError('Argument "image_position" must have length 3.')
+    if not isinstance(image_orientation, Sequence):
+        raise TypeError('Argument "image_orientation" must be a sequence.')
+    if len(image_orientation) != 6:
+        raise ValueError('Argument "image_orientation" must have length 6.')
+    if not isinstance(pixel_spacing, Sequence):
+        raise TypeError('Argument "pixel_spacing" must be a sequence.')
+    if len(pixel_spacing) != 2:
+        raise ValueError('Argument "pixel_spacing" must have length 2.')
+
+    x_offset = float(image_position[0])
+    y_offset = float(image_position[1])
+    z_offset = float(image_position[2])
+    translation = np.array([x_offset, y_offset, z_offset])
+
+    rotation = create_rotation_matrix(image_orientation)
+    # Column direction (spacing between rows)
+    column_spacing = float(pixel_spacing[0])
+    # Row direction (spacing between columns)
+    row_spacing = float(pixel_spacing[1])
+    rotation[:, 0] *= row_spacing
+    rotation[:, 1] *= column_spacing
+    rotation[:, 2] *= spacing_between_slices
+    inv_rotation = np.linalg.inv(rotation)
+
+    # 4x4 transformation matrix
+    return np.row_stack(
+        [
+            np.column_stack([
+                inv_rotation,
+                -np.dot(inv_rotation, translation)
+            ]),
+            [0.0, 0.0, 0.0, 1.0]
+        ]
+    )
+
+
+class PixelToReferenceTransformer(object):
+
+    """Class for transforming pixel indices to reference coordinates.
```

Comment:
I think it would really help users to add a clarifying `Note` here about the two definitions of the image coordinate system and how these four classes relate to them, what do you think?